src/main/java/io/microprofile/config/spi/ConfigSource.java

```diff
@@ -59,6 +64,19 @@
 	 * @param propertyName the property name
 	 * @return the property value 
 	 */
-	String getProperty(String propertyName);
+	String getPropertyValue(String propertyName);
+	
+	/**
+     * The name of the config might be used for logging or analysis of configured values.
+     *
+     * @return the 'name' of the configuration source, e.g. 'property-file mylocation/myproperty.properties'
```

Comment:
@ebullient It's all a matter of terminology. A good design usually finds well defined terms with a clear meaning and behaviour. 
In that case I proposed:
Config (the whole Configuration of an application)
 -> [consists of] n ConfigProperties 
  -> [where each ConfigProperty consists of a]  Config Property Key and a Config Property Value